File: tools/android_host_bridge/org/oraclarva/mobile/NativeBridge.java
package org.oraclarva.mobile;

final class NativeBridge {
    static {
        System.loadLibrary("oraclarva_android");
    }

    private NativeBridge() {}

    static native long nativeCreate(String repeatFixturePath, String spatialFixturePath);
    static native void nativeDestroy(long handle);
    static native void nativeReset(long handle);
    static native int nativeAdvance(
            long handle,
            double posteriorTouchIntensity,
            boolean lightEnabled,
            double originXM,
            double originYM,
            double originZM,
            double valueAtOriginWM2,
            double gradientXWM3,
            double gradientYWM3,
            double gradientZWM3,
            double temporalRateWM2S,
            double lowerBoundWM2,
            double upperBoundWM2,
            int steps);
    static native int[] nativeRenderCounts(long handle);
    static native int nativeReadFrame(
            long handle,
            double[] stateValues,
            float[] renderVertices,
            int[] renderIndices);
}

File: android/app/src/main/kotlin/org/oraclarva/mobile/NativeBridge.kt
package org.oraclarva.mobile

internal class NativeBridge private constructor() {
    companion object {
        init {
            System.loadLibrary("oraclarva_android")
        }

        @JvmStatic
        external fun nativeCreate(
            repeatFixturePath: String,
            spatialFixturePath: String,
        ): Long

        @JvmStatic
        external fun nativeDestroy(handle: Long)

        @JvmStatic
        external fun nativeReset(handle: Long)

        @JvmStatic
        external fun nativeAdvance(
            handle: Long,
            posteriorTouchIntensity: Double,
            lightEnabled: Boolean,
            originXM: Double,
            originYM: Double,
            originZM: Double,
            valueAtOriginWM2: Double,
            gradientXWM3: Double,
            gradientYWM3: Double,
            gradientZWM3: Double,
            temporalRateWM2S: Double,
            lowerBoundWM2: Double,
            upperBoundWM2: Double,
            steps: Int,
        ): Int

        @JvmStatic
        external fun nativeRenderCounts(handle: Long): IntArray

        @JvmStatic
        external fun nativeReadFrame(
            handle: Long,
            stateValues: DoubleArray,
            renderVertices: FloatArray,
            renderIndices: IntArray,
        ): Int
    }
}
